### PR TITLE
Add documentation

### PR DESCRIPTION
**Pull request by jjpr-mit**
Added documentation to README. Added formal-ish specification for the language-agnostic core standard.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,6 +1,229 @@
...
+BrainIO is intended to find a balance between the effort it requires of the consumer of data and the effort it requires of the provider of data, and to minimize both where possible.  
+### Terms and Concepts
+The [BrainIO Format Specification](docs/SPECIFICATION.md) defines three types of entity:  
+* A Stimulus Set
+* A Data Assembly
+* A BrainIO Catalog
+
+A **Stimulus Set** consists of:
+* a CSV file of metadata
+* a zip file of stimuli
```
[the remark this answers is not in the document]
Comment (jjpr-mit) [reply README.md:32]: I thought I remembered zip being stylized as lowercase officially.  I remembered wrong:  https://pkware.cachefly.net/webdocs/casestudies/APPNOTE.TXT.  Oops.  Definitely officially uppercase "ZIP".  Also:  https://datatracker.ietf.org/doc/html/rfc4180.  Closest thing to official, uses uppercase "CSV".  I'll make all upper.

File: README.md [hunk 2]
```diff
@@ -1,6 +1,229 @@
...
+* Other people will know less than you about what the data means and how to use it
+* **Future you** will know less than **present you** about what the data means and how to use it
+* If you can help that knowledge accompany the data on its travels, you can expand the contexts in which it is useful 
+#### To Adopt A Consistent Format
+* Everyone stores their data differently.  One person uses a single monolithic CSV file, another uses five separate CSV files;  one keeps data in the proprietary binary format the instrument produces, yet another uses a fifteen year old version of MATLAB  
+* In principle data in one style of organization can be transformed to any other style of organization
+* In practice people are rarely willing to spend the time to translate multiple conventions to the same format just so they can use data from multiple sources  
+#### For A Fair Exchange Of Effort
+BrainIO is intended to find a balance between the effort it requires of the consumer of data and the effort it requires of the provider of data, and to minimize both where possible.  
+### Terms and Concepts
```
[the remark this answers is not in the document]
Comment (jjpr-mit) [reply README.md]: Done in 5a14826fecfc23e5df59d1ca512b8bad6b6ee2f4

File: README.md [hunk 3]
```diff
@@ -1,6 +1,229 @@
...
+### Why use BrainIO?
+#### To Safeguard The Value Of Data You Have Collected
+* All data decays in usefulness as it gets farther from the place and time where it was collected
+* Other people will know less than you about what the data means and how to use it
+* **Future you** will know less than **present you** about what the data means and how to use it
+* If you can help that knowledge accompany the data on its travels, you can expand the contexts in which it is useful 
+#### To Adopt A Consistent Format
+* Everyone stores their data differently.  One person uses a single monolithic CSV file, another uses five separate CSV files;  one keeps data in the proprietary binary format the instrument produces, yet another uses a fifteen year old version of MATLAB  
+* In principle data in one style of organization can be transformed to any other style of organization
+* In practice people are rarely willing to spend the time to translate multiple conventions to the same format just so they can use data from multiple sources
```
[the remark this answers is not in the document]
Comment (jjpr-mit) [reply README.md]: Chris and I decided to trim that section down a lot, it was too long.  How strongly do you feel about adding this?

File: README.md [hunk 4]
```diff
@@ -1,6 +1,229 @@
...
+* **Future you** will know less than **present you** about what the data means and how to use it
+* If you can help that knowledge accompany the data on its travels, you can expand the contexts in which it is useful 
+#### To Adopt A Consistent Format
+* Everyone stores their data differently.  One person uses a single monolithic CSV file, another uses five separate CSV files;  one keeps data in the proprietary binary format the instrument produces, yet another uses a fifteen year old version of MATLAB  
+* In principle data in one style of organization can be transformed to any other style of organization
+* In practice people are rarely willing to spend the time to translate multiple conventions to the same format just so they can use data from multiple sources  
+#### For A Fair Exchange Of Effort
+BrainIO is intended to find a balance between the effort it requires of the consumer of data and the effort it requires of the provider of data, and to minimize both where possible.  
+### Terms and Concepts
+The [BrainIO Format Specification](docs/SPECIFICATION.md) defines three types of entity:
```
[the remark this answers is not in the document]
Comment (jjpr-mit) [reply README.md]: [[citation needed]] No answer with citation shows up here:  https://english.stackexchange.com/questions/5539/types-of-things-vs-types-of-thing

But it's moot anyway, we rewrote this section.

File: README.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (jjpr-mit) [reply README.md]: Fixed in 5a14826fecfc23e5df59d1ca512b8bad6b6ee2f4.

File: README.md [hunk 5]
```diff
@@ -1,6 +1,229 @@
...
+* A Data Assembly
+* A BrainIO Catalog
+
+A **Stimulus Set** consists of:
+* a CSV file of metadata
+* a zip file of stimuli
+* a few rules for how the metadata are organized (like, include a column called "image_id")
+
+A **Data Assembly** consists of:
+* a netCDF file containing numeric data and metadata
```
[the remark this answers is not in the document]
Comment (jjpr-mit) [reply README.md]: Added link.

File: README.md [hunk 6]
```diff
@@ -1,6 +1,229 @@
...
+>>> hvm_stim = get_stimulus_set("dicarlo.hvm")
+>>> hvm_stim
+        id                             background_id         s  ... rxy_semantic ryz_semantic        rxz
+0        1  ecd40f3f6d7a4d6d88134d648884e0b9b364efc9  1.000000  ...    90.000000    -0.000000   0.000000
+1        2  006d66c207c6417574f62f0560c6b2b40a9ec5a1  1.000000  ...    -0.000000    -0.000000   0.000000
+...
+5758  5759  2363c8c3859603ee9db7f83dd7ba5b6989154258  0.791250  ...   -75.159000    12.283000  29.466000
+5759  5760  a7fc954d2bd08fc2c4d3ade347f41a03a5850131  1.050000  ...   -67.467580     7.598457 -25.152591
+
+[5760 rows x 18 columns]
```
[the remark this answers is not in the document]
Comment (jjpr-mit) [reply README.md:97]: Added in 5a14826fecfc23e5df59d1ca512b8bad6b6ee2f4.

File: README.md [hunk 7]
```diff
@@ -1,6 +1,229 @@
...
+```shell
+conda create -n my-project python==3.8
+conda activate my-project
+pip install -e ~/projects/my-project
+```
+You should now be able to use packaging scripts (with the argument 
+`catalog_name=my_project`) to add to your catalog.  
+## Getting Help
+
+If you have questions, [get in touch with us]("mailto:jjpr@mit.edu,mferg@mit.edu?subject=Brain-Score submission).
```
[the remark this answers is not in the document]
Comment (jjpr-mit) [reply README.md]: Oops.  Fixed in 5a14826fecfc23e5df59d1ca512b8bad6b6ee2f4.

File: docs/SPECIFICATION.md [hunk 8]
```diff
@@ -0,0 +1,40 @@
...
+    * The CSV file may contain zero or more columns containing metadata values pertaining to the stimuli described.  
+## Data Assembly
+A Data Assembly is an n-dimensional array of data collected during an experiment, the metadata which provide meaning to those data and a reference to the Stimulus Set used in the experiment.  
+* A Data Assembly must be associated with a string identifier.  The identifier must be unique among the Data Assemblies in any BrainIO Catalog which contains it.  
+* The data and metadata must be contained in a netCDF file.  
+* Only one netCDF Variable in the file may contain experimental data.  All other Variables will be treated as metadata.  
+* The netCDF file must have a global attribute named `identifier` which contains the Data Assembly's identifier
+* The netCDF file must have a global attribute named `stimulus_set_identifier` which contains the identifier of the BrainIO Stimulus Set used in the experiment from which the data were collected.  
+* The netCDF file must contain a dimension named `presentation` and a metadata Variable with that dimension named `image_id`.
```
[the remark this answers is not in the document]
Comment (jjpr-mit) [reply docs/SPECIFICATION.md:25]: Yes, but it's outside the scope of this PR.

File: README.md [hunk 9]
```diff
@@ -1,6 +1,229 @@
...
+
+This registers the catalog with the Python interpreter in your environment 
+so that the BrainIO tools can use it.  
+
+### Set Up To Add To A Project That Provides a BrainIO Catalog
+
+If you intend to package Stimulus Sets And Data Assemblies and add them 
+to a project's catalog, instead of installing the project as above, you will need to clone the repository for that project and 
+install it in your environment in editable mode.  In the example below we set up to 
+add to `brianio-dicarlo`'s BrainIO Catalog:
```
[the remark this answers is not in the document]
Comment (jjpr-mit) [reply README.md]: Fixed in 87db1c2edaadc639550da03154bf2c9f295e5bb4.

File: docs/SPECIFICATION.md [hunk 10]
```diff
@@ -0,0 +1,40 @@
+# BrainIO Format Specification v0.1
```
[the remark this answers is not in the document]
Comment (jjpr-mit) [reply docs/SPECIFICATION.md]: Well no actually but "start doing real releases with version numbers" has been on my to-do list so long I was thinking of forcing myself to do it.  But I've changed my mind.  Removed in 87db1c2edaadc639550da03154bf2c9f295e5bb4.

File: docs/SPECIFICATION.md [hunk 11]
```diff
@@ -0,0 +1,40 @@
...
+    * The CSV file must contain a column named `image_id`.  Each metadata row in the CSV must have an alphanumeric string which is unique within the Stimulus Set in the `image_id` column.  This string is the identifier for the stimulus which is described by the metadata in that row. 
+    * The CSV file must contain a column named `filename`.  The `filename` value for each metadata row is the name of the file in the zip archive (or full file path if the zip archive is organized in directories) which corresponds to the stimulus described by that row.  
+    * The CSV file may contain zero or more columns containing metadata values pertaining to the stimuli described.  
+## Data Assembly
+A Data Assembly is an n-dimensional array of data collected during an experiment, the metadata which provide meaning to those data and a reference to the Stimulus Set used in the experiment.  
+* A Data Assembly must be associated with a string identifier.  The identifier must be unique among the Data Assemblies in any BrainIO Catalog which contains it.  
+* The data and metadata must be contained in a netCDF file.
```
[the remark this answers is not in the document]
Comment (jjpr-mit) [reply docs/SPECIFICATION.md]: Fixed in 87db1c2edaadc639550da03154bf2c9f295e5bb4.

File: docs/glossary.md [hunk 12]
```diff
@@ -0,0 +1,19 @@
...
+
+## Other Terms Used
+
+**Data**:  information, usually the numeric result of physical measurements performed in the context of an experiment, stored digitally
+
+**Metadata**:  information that describes other information
+
+**Repository**:  a set of files and directories managed by a version control system such as Git;  cloud services such as GitHub host copies of repositories to make them easier to share
+
+**Pull Request**:  a mechanism offered by cloud version control system providers by which users can offer code to other users
```
[the remark this answers is not in the document]
Comment (jjpr-mit) [reply docs/glossary.md]: Fixed in 87db1c2edaadc639550da03154bf2c9f295e5bb4.

File: README.md [hunk 13]
```diff
@@ -49,15 +49,15 @@ These are described below.  We recommend installing in a managed Python environm
...
     1. Activate your environment
     1. `pip install git+https://github.com/dicarlolab/brainio-dicarlo.git`
     * This registers the catalog with the Python interpreter in your environment 
 so that the BrainIO tools can use it.  
 
 ## Install BrainIO As A Provider Of Packages
 
-If you intend to package Stimulus Sets And Data Assemblies and add them to a project's catalog, instead of installing the project as above, you will need to clone the repository for that project and install it in your environment in editable mode.  In the example below we set up to add to `brianio-dicarlo`'s BrainIO Catalog:  
+If you intend to package Stimulus Sets And Data Assemblies and add them to a project's catalog, instead of installing the project as above, you will need to clone the repository for that project and install it in your environment in editable mode.  In the example below we set up to add to the BrainIO Catalog provided by [a project called `brianio-dicarlo`](https://github.com/dicarlolab/brainio-dicarlo):
```
Comment (mschrimpf) [inline README.md:60]: I still don't know what brianio is

File: README.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (mschrimpf) [reply README.md]: I'm all for trimming down! You're now pointing to hdf5 so that's good

**Review by mschrimpf — APPROVED** [review]
looks good to me once you fix brian

File: README.md (hunk 13 above)
Comment (jjpr-mit) [reply README.md]: He's just this guy I know.  Figured I'd give him a shoutout, but fine, have it your way.  😆

**Pull request merged**